keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,1114 @@
+    - [Availability](#availability)
+    - [Automatic creation](#automatic-creation)
+    - [Manual creation](#manual-creation)
+    - [Network Migration](#network-migration)
+  - [PodNetwork Controller](#podnetwork-controller)
+  - [Feature gate](#feature-gate)
+  - [Attaching PodNetwork to a Pod](#attaching-podnetwork-to-a-pod)
+    - [Static validations](#static-validations)
+    - [Active validations](#active-validations)
+    - [Auto-population](#auto-population)
+    - [Status](#status)
+    - [DRA integration (alternative)](#dra-integration-alternative)
+  - [API server changes](#api-server-changes)
+  - [Scheduler changes](#scheduler-changes)
+  - [Endpointslice controller changes](#endpointslice-controller-changes)
+  - [Kubelet changes](#kubelet-changes)
+  - [CRI changes](#cri-changes)
+    - [Pod Creation](#pod-creation)
+    - [Pod Status](#pod-status)
+    - [CNI API](#cni-api)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Alpha](#alpha)
+    - [Beta](#beta)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+One of the main requirements for Kubernetes networking today is to enable
+connectivity between `Pods` in a cluster. This facility satisfies a large number
+of Kubernetes users, but it isn't sufficient for some important cases:
+
+* applications leveraging different isolated networks exposed through different
+  interfaces
+* applications leveraging performance-oriented interfaces (e.g. `AF_XDP`,
+  `memif`, `SR-IOV`)
+* applications requiring support for protocols not yet supported by Kubernetes
+
+Here we propose a solution that enables multiple network interfaces to be
+defined in the `Pod` specification to provide a standard and clear mechanism to
+implement these more complex configurations.
+
+## Motivation
+
+We want to have a common API allowing us to define a catalog of different
+networks in the Kubernetes cluster. It would allow attaching a pod to one or
+several networks via a given type of interface depending on its connectivity
+or performance needs.
+
+### Goals
+
+* Define user stories and requirements for the Multi-Network effort in Kubernetes.
+* Introduce API object to kubernetes describing networks Pod can attach to.
+* Evolve current Kubernetes networking model to support multiple networks,
+defining the new model in a backwards compatible way.
+* Integrate this API with the existing core objects.
+* Define “reference implementation” that can be used in Kubernetes CI.
+* Define conformance tests.
+
+
+### Non-Goals
+
+Define the CNI implementation.
```

Comment:
Non-goal:
* To interfere with routing inside the pod

With multiple interfaces routing becomes an issue. It should be left to implementations. There should _not_ be a requirement to not touch the default route for instance.

Reply:
updated